src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -0,0 +1,603 @@
+
+    def work_for_vts_rsc_outline(vts, ctx, key):
+      # Double check the cache before beginning compilation
+      hit_cache = self.check_cache(vts, counter)
+
+      if not hit_cache:
+        is_incremental = self.should_compile_incrementally(vts, ctx)
+        if not is_incremental:
+          # Purge existing analysis file in non-incremental mode.
+          safe_delete(ctx.analysis_file)
+          # Work around https://github.com/pantsbuild/pants/issues/3670
+          safe_rmtree(ctx.classes_dir)
+
+        cp_entries = []
+
+        # Include the current machine's jdk lib jars. This'll blow up remotely.
+        # We need a solution for that.
+        # Probably something to do with https://github.com/pantsbuild/pants/pull/6346
+        distribution = JvmPlatform.preferred_jvm_distribution([ctx.target.platform], strict=True)
+        jvm_lib_jars_abs = distribution.find_libs(['rt.jar', 'dt.jar', 'tools.jar'])
+        cp_entries.extend(jvm_lib_jars_abs)
+
+        classpath_abs = self._zinc.compile_classpath(
+          'rsc_outline_classpath',
+          ctx.target,
+          extra_cp_entries=self._extra_compile_time_classpath)
+        classpath_rel = relpathify(classpath_abs)
+        cp_entries.extend(classpath_rel)
+
+        ctx.ensure_output_dirs_exist()
+        
+        tgt, = vts.targets
+        with Timer() as timer:
+          # Step 1: Convert classpath to SemanticDB
+          # ---------------------------------------
+          scalac_classpath_path_entries_abs = self.tool_classpath('workaround-metacp-dependency-classpath')
+          scalac_classpath_path_entries = relpathify(scalac_classpath_path_entries_abs)
+          rsc_outline_dir = relpathalone(ctx.rsc_outline_dir)
+          args = [
+            # NB: Without this setting, rsc will be missing some symbols
+            #     from the scala library.
+            '--include-scala-library-synthetics', # TODO generate these once and cache them
+            # NB: We need to add these extra dependencies in order to be able
+            #     to find symbols used by the scalac jars.
+            '--dependency-classpath', os.pathsep.join(scalac_classpath_path_entries),
+            # NB: The directory to dump the semanticdb jars generated by metacp.
+            # TODO: break this out into a separate job so that we can apply
+            #       it once per 3rdparty target.
+            '--out', rsc_outline_dir,
+            os.pathsep.join(cp_entries),
+          ]
+          metacp_wu = self._runtool(
+            'scala.meta.cli.Metacp',
+            'metacp',
+            args,
+            distribution,
+            tgt=tgt,
+            input_files=(scalac_classpath_path_entries + classpath_rel),
+            output_dir=rsc_outline_dir)
+          metacp_stdout = stdout_contents(metacp_wu)
+          metacp_result = json.loads(metacp_stdout)
+          metai_classpath = []
+          
+          def desandboxify_pantsd_loc(path):
```

Comment:
nit: assuming `pantsd` means workdir here, can we use `workdir` or `_pants_d_`?

Reply:
I tried that and it didn't work with tests because of how tests place their temp workdirs. I have an idea for a different approach now though.